test/functional/admin/editions_controller_test.rb

```diff
@@ -81,6 +81,25 @@ class Admin::EditionsControllerTest < ActionController::TestCase
     assert_redirected_to edit_admin_publication_path(draft_edition.reload)
   end
 
+  test "failing to revise an edition should surface any ActiveRecord errors" do
+    error_message = "Validation failed: Summary can't be blank"
+    I18n.stubs(:t).returns("foo") # for any other calls to `I18n.t`
+    I18n.stubs(:t).with(:"activemodel.errors.messages.record_invalid", anything).returns(error_message)
+
+    published_consultation = create(:closed_consultation) do |consultation|
+      Object.stub_const("InvalidRecord", Class.new { include ActiveModel::Model }) do
+        exception = ActiveRecord::RecordInvalid.new(InvalidRecord.new)
+        trait = Edition::Traits::Trait.any_instance.stubs(:process_associations_after_save).raises(exception)
+        consultation.stubs(:traits).returns([trait])
```

Comment:
I don't really like the amount of stubbing we're doing here... would it be possible to do this using a factory? You can always save an invalid edition by setting the `validate: false` override when saving the model (and `build` rather than `create` on the factory), if that's the problem you're working around

Reply:
Thanks for the tip - I think I tried and gave up on a `validate: false` option (perhaps because the validation needs to be disabled on the ConsultationPublicFeedback, rather than the Consultation itself). Have given it another go, much cleaner now 👍